ionoscloud/resource_server.go

```diff
@@ -531,16 +536,30 @@ func resourceServerCreate(ctx context.Context, d *schema.ResourceData, meta inte
 		return diags
 	}
 
+	labels := getLabels(d.Get("label"))
+	for _, label := range labels {
+		labelKey := label["key"]
+		labelValue := label["value"]
+		labelResource := ionoscloud.LabelResource{
+			Properties: &ionoscloud.LabelResourceProperties{Key: &labelKey, Value: &labelValue},
+		}
+		_, apiResponse, err := client.LabelsApi.DatacentersServersLabelsPost(ctx, datacenterId, *postServer.Id).Label(labelResource).Execute()
+		logApiRequestTime(apiResponse)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("error creating label: (%w)", err))
+		}
+	}
+
 	// get additional data for schema
-	serverReq, apiResponse, err = client.ServersApi.DatacentersServersFindById(ctx, d.Get("datacenter_id").(string), *postServer.Id).Depth(3).Execute()
+	serverReq, apiResponse, err = client.ServersApi.DatacentersServersFindById(ctx, datacenterId, *postServer.Id).Depth(3).Execute()
```

Comment:
hi, just a remark here: from what I understand, we use a GET Servers with a depth of 3 (pretty big) in order to get this server's Nics as well.

This is just a suggestion, but to improve performance, we could instead do a GET Servers with a depth of 0, and then use a GET on the NicsApi, again with a depth of 0 to get the "firstNicItem" variable below

Reply:
@avirtopeanu-ionos

> we use a GET Servers with a depth of 3 (pretty big)

I know and I agree, the depth is way bigger than what we actually need, but unfortunately there are a lot of mini-problems in the provider, like this one, so for now, I only wrote them to take care of them in the future and I focused on what was needed for this PR. In the future we need to modify this.